lib/helpers.bash

```diff
@@ -849,3 +849,30 @@ then
     fi
   }
 fi
+
+# `_bash-it-find-in-ancestor` uses the shell's ability to run a function in
+# a subshell to simplify our search to a simple `cd ..` and `[[ -r $1 ]]`
+# without any external dependencies. Let the shell do what it's good at.
+function _bash-it-find-in-ancestor() (
+	about 'searches parents of the current directory for the specified file name'
+	group 'helpers'
+	param '1: name of file or folder to search for'
+	param '2: alternate name to search for'
+	returns '0: prints path to matching parent directory to stdout'
+	returns '1: no match found'
+	returns '2: improper usage of shell builtin' # uncommon
+	example '_bash-it-find-in-ancestor .git'
+	example '_bash-it-find-in-ancestor Makefile'
```

Comment:
This is coming along great ...

If we're *really* going to accept a list of files/dirs and stop on first match, then lets make the docs more clear on that ... Maybe something along the lines of:

```
about 'searches parents of the current directory for the specified file/folder names, stopping on first match'
param '*: names of files or folders to search for, in priority order'
returns '0: prints path of first matching parent directory to stdout'
```

Feel free to clean up the copy as you see fit, but you get the idea.

Reply:
How's this look now?